spot-price-monitor.py

```diff
@@ -112,26 +165,51 @@ def update_spot_price_metrics(metric, prices):
     v1 = client.CoreV1Api()
     ec2 = boto3.client('ec2', args.region)
     start_http_server(args.metrics_port)
+    o = None
 
     s = Gauge('aws_spot_price_dollars_per_hour',
               'Reports the AWS spot price of node types used in the cluster',
               ['instance_type', 'availability_zone']
               )
+    if args.ondemand:
+        o = Gauge('aws_on_demand_dollars_per_hour',
+                  'Reports the AWS ondemand of node types used in the cluster',
+                  ['instance_type', 'availability_zone']
+                 )
+
     error = Counter('aws_spot_price_request_errors',
                     'Reports errors while calling the AWS api.',
                     ['code']
                     )
 
+    last_ondemand_update = 0
     backoff_multiplier = 1
+    spot_prices = []
     while(True):
         zones = get_zones_from_k8s(v1)
         try:
             types = get_instance_types_from_k8s(v1, args.spot_label)
             spot_prices = get_spot_prices(ec2, types, zones, args.products)
             backoff_multiplier = 1
         except ClientError as e:
-            error.label(code=e.response['Error']['Code']).inc()
+            error.labels(code=e.response['Error']['Code']).inc()
             if e.response['Error']['Code'] == 'RequestLimitExceeded':
                 backoff_multiplier *= 2
         update_spot_price_metrics(s, spot_prices)
-        sleep(args.scrape_interval * backoff_multiplier)
+
+        # refresh ondemand prices each hour
```

Comment:
This comment says "each hour" but it looks like the time interval is actually one day

Reply:
Fixed